lib/active_model/serializer.rb

```diff
@@ -177,6 +177,10 @@ def json_key
       end
     end
 
+    def as_json
+      self.class.adapter.new(self).to_json
```

Comment:
Also update the adapter call to use ```as_json``` method (link)[https://github.com/rails-api/active_model_serializers/blob/master/lib/active_model/serializer/adapter.rb#L22]